js/popup.js

```diff
@@ -1,228 +1,99 @@
-      $("#message").show();
-      var msg = "Invalid request!";
-      $("#message").html("<p>" + msg + "</p>");
-      $("#login").show();
-      $("#main").show();
-    }
-    else{
-      $("#login").hide();
-      $("#user_box").hide();
-      $("#message").show();
-      var msg = "Error occured!";
-      $("#message").html("<p>" + msg + "</p>");
-      $("#main").show();
-    }
-  })
-  .fail( function(res) {
-    console.log("Error occured while fetching data-1-3!");
-    $("#loader").hide();
-    /* NOTE: No need to load apps_content into main b'coz it's already 
-       loaded at .failure of GET'/chrome-ext/check-cache/' */
-    var msg = "Network connection error!";
-    $("#message").html("<p>" + msg + "</p>");
-    $("#login").show();
-    $("#main").show();
-    chrome.browserAction.setIcon({path: "../images/icon_inactive.png"});
-  });
-}
+    $login.hide();
+    $main.hide();
 
-  if(typeof localStorage['cache_key'] == 'undefined') {
-    $.get(Domain + '/chrome-ext/check-cache/', {}, function(res){
-      if(!(res == 'FAILURE')) {
-        $('#main').html(res);
-        var new_key = $('input[name="cache_key"]').val();
-        if(!(new_key == 'undefined')) {
-          localStorage['cache_key'] = new_key;
-          localStorage['apps_content'] = res;
-        }
-      }
-      else {
-        /* Fetch data from localstorage */
-        console.log("Error occured while fetching data!");
-        var html_content = localStorage['apps_content'];
-        $('#main').html(html_content);
-      }
-    })
-    .fail( function(res) {
-      /* Fetch data from localstorage */
-      console.log("Error occured while fetching data!");
-      var html_content = localStorage['apps_content'];
-      $('#main').html(html_content);
-    });
-  }
-  else {
-    $.get( Domain + '/chrome-ext/check-cache/', {'cache_key': localStorage['cache_key']}, function(res){
-      console.log("success!!");
-      if(!(res == 'FAILURE')) {
-        if(typeof res.cache_key == 'undefined') {
-          $('#main').html(res);
-          var new_key = $('input[name="cache_key"]').val();
-          if(!(typeof new_key == 'undefined')) {
-            localStorage['cache_key'] = new_key;
-            localStorage['apps_content'] = res;
-          }
-        }
-        else {
-          /* Fetch data from localstorage */
-          console.log("Already have updated version!");
-          var html_content = localStorage['apps_content'];
-          $('#main').html(html_content);
-        }
-      }
-      else {
-        /* Fetch data from localstorage */
-        console.log("Error occured while fetching data!");
-        var html_content = localStorage['apps_content'];
-        $('#main').html(html_content);
-      }
-    })
-    .fail( function(res) {
-      /* Fetch data from localstorage */
-      console.log("Error occured while fetching data!");
-      var html_content = localStorage['apps_content'];
-      $('#main').html(html_content);
-    });
-  }
```

Comment:
We've written this local storage part to fetch the recent app listing data from the backend irrespective of the state of the plugin on the client side for maintaining consistency. This is similar of versioning

So if you are fetching the app listing template & data dynamically from the backend then you have to implement this kind of versioning.